### PR TITLE
Fix colormap utils error suggestion code and add a test

### PR DESCRIPTION
**Pull request by goanpeca**
# Description

## Type of change

- [x] Bug-fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# References

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [x] example: all tests pass with my change
- [ ] example: I check if my changes works with both PySide and PyQt backends
      as there are small differences between the two Qt bindings.  

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/developers/translations.html).

### CODE REVIEW

**Review by jni — APPROVED** [review]

File: napari/utils/colormaps/_tests/test_colormaps.py [hunk 1]
```diff
@@ -161,22 +161,36 @@ def test_mpl_colormap_exists():
...
+        ('light_blues', 'light blues'),  # Vispy
+    ],
+)
+def test_colormap_error_suggestion(name, display_name):
     """
     Test that vispy/mpl errors, when using `display_name`, suggest `name`.
     """
-    name = '"twilight_shifted"'
-    display_name = 'twilight shifted'
     with pytest.raises(KeyError) as excinfo:
```
Comment (Czaki) [inline napari/utils/colormaps/_tests/test_colormaps.py]:
```suggestion
    with pytest.raises(KeyError, match=rf"{display_name}.*you might want to use.*{name}"):
```

File: napari/utils/colormaps/_tests/test_colormaps.py [hunk 2]
```diff
@@ -161,22 +161,36 @@ def test_mpl_colormap_exists():
...
         vispy_or_mpl_colormap(display_name)
 
-    assert name in str(excinfo.value)
+    error_string = str(excinfo.value)
+    assert 'you might want to use' in error_string
+    assert name in error_string and display_name in error_string
```
Comment (Czaki) [inline napari/utils/colormaps/_tests/test_colormaps.py]:
```suggestion
```

File: napari/utils/colormaps/_tests/test_colormaps.py [hunk 3]
```diff
@@ -161,22 +161,36 @@ def test_mpl_colormap_exists():
...
+
 
-    wrong_name = 'foobar'
+def test_colormap_error_from_inexistent_name():
+    """
+    Test that vispy/mpl errors when using a wrong name.
+    """
+    name = 'foobar'
     with pytest.raises(KeyError) as excinfo:
```
Comment (Czaki) [inline napari/utils/colormaps/_tests/test_colormaps.py]:
```suggestion
    with pytest.raises(KeyError, match=rf"{name}.*Recognized colormaps are"):
```

File: napari/utils/colormaps/_tests/test_colormaps.py [hunk 4]
```diff
@@ -161,22 +161,36 @@ def test_mpl_colormap_exists():
...
-        vispy_or_mpl_colormap(wrong_name)
+        vispy_or_mpl_colormap(name)
 
-    assert name in str(excinfo.value)
+    error_string = str(excinfo.value)
+    assert 'Recognized colormaps are' in error_string
+    assert name in error_string
```
Comment (Czaki) [inline napari/utils/colormaps/_tests/test_colormaps.py]:
```suggestion
```

**Review by Czaki — COMMENTED** [review]
I prefer to use match instead string assertions, but I'm ok with both.

**Comment by goanpeca** [comment]
>I prefer to use match instead string assertions, but I'm ok with both.

TIL :) that is pretty cool, thanks for the suggestion!

**Review by Czaki — APPROVED** [review]

**Pull request merged**